### PR TITLE
fix order of related products

### PR DESCRIPTION
**Pull request by zhuwu**
My pull request intends to fix the order of related products returned from the method `relations_for_relation_type`.

The original implementation generated SQL like the following:

```
SELECT "spree_products".* FROM "spree_products" WHERE "spree_products"."id" IN 
  (
    SELECT "spree_relations"."related_to_id" FROM "spree_relations" 
      WHERE "spree_relations"."relatable_id" = $1 
      AND "spree_relations"."relatable_type" = $2 
      AND "spree_relations"."relation_type_id" = $3  
      ORDER BY "spree_relations"."position" ASC
  )
  AND "spree_products"."deleted_at" IS NULL 
  AND "spree_products"."deleted_at" IS NULL 
  AND (spree_products.available_on IS NOT NULL) 
  AND (spree_products.available_on <= '2016-08-15 02:26:05.897318')  
```

However, the logic was not correct, because the order by position logic was put inside the sub query. The result of outer query will not follow the order specified in inner query. There are tons of discussions online as well. You may refer to [http://stackoverflow.com/questions/2101908/is-order-by-clause-allowed-in-a-subquery](http://stackoverflow.com/questions/2101908/is-order-by-clause-allowed-in-a-subquery) or [http://dba.stackexchange.com/questions/82930/database-implementations-of-order-by-in-a-subquery](http://dba.stackexchange.com/questions/82930/database-implementations-of-order-by-in-a-subquery)

In short, the order by clause has to be put on the outermost query to guarantee the correct result. Thus, I alter the implementation to use join instead in this pull request. The generated SQL will look like this after my modification:

```
SELECT "spree_products".* FROM "spree_products" INNER JOIN spree_relations 
  on spree_products.id = spree_relations.related_to_id 
  and spree_relations.related_to_type = 'Spree::Product' 
  WHERE "spree_relations"."relatable_id" = 35 
  AND "spree_relations"."relation_type_id" = 1 
  AND "spree_products"."deleted_at" IS NULL 
  AND "spree_products"."deleted_at" IS NULL 
  AND (spree_products.available_on IS NOT NULL) 
  AND (spree_products.available_on <= '2016-08-15 06:51:50.101095')  
  ORDER BY spree_relations.position
```

### CODE REVIEW

**Comment by octave** [comment]
The order seems to still be broken. Does anyone figured it out?

**Comment by kennyadsl** [comment]
Closing this one for inactivity, thanks anyway!
